src/components/PopupWithForm.js

```diff
@@ -16,9 +17,13 @@ class PopupWithForm extends Popup {
     return formValues;
   }
 
-  submit(e) {
+  submit(e) { // handles all submitting stuff
     e.preventDefault();
+    const buttonLabel = this._popupElement.querySelector(".modal__save").innerText;
```

Comment:
`this._popupElement.querySelector(".modal__save")` is searched 3 times here. Please, define a variable or a class field for that  because searching is a very **hard** operation for the browser engine